pkg/kp/dsstore/consul_store.go

```diff
@@ -119,6 +120,24 @@ func (s *consulStore) innerCreate(
 	return ds, nil
 }
 
+func (s *consulStore) Delete(id fields.ID) error {
+	dsPath, err := s.dsPath(id)
+	if err != nil {
+		return err
+	}
+
+	_, _, err = s.Get(id)
```

Comment:
this probably isn't necessary, it's okay if we try to delete something that doesn't exist and i'm not sure how that would happen